kernel/aster-nix/src/process/process/mod.rs

```diff
@@ -98,31 +232,70 @@ impl Process {
         sig_dispositions: Arc<Mutex<SigDispositions>>,
         resource_limits: ResourceLimits,
         nice: Nice,
-    ) -> Self {
+    ) -> Arc<Self> {
         let children_pauser = {
             // SIGCHID does not interrupt pauser. Child process will
             // resume paused parent when doing exit.
             let sigmask = SigMask::from(SIGCHLD);
             Pauser::new_with_mask(sigmask)
         };
 
-        Self {
-            pid,
-            threads: Mutex::new(threads),
-            executable_path: RwLock::new(executable_path),
-            process_vm,
-            children_pauser,
-            status: Mutex::new(ProcessStatus::Uninit),
-            parent: Mutex::new(parent),
-            children: Mutex::new(BTreeMap::new()),
-            process_group: Mutex::new(Weak::new()),
-            file_table,
-            fs,
-            umask,
-            sig_dispositions,
-            resource_limits: Mutex::new(resource_limits),
-            nice: Atomic::new(nice),
-        }
+        let process = Arc::new_cyclic(|process_ref: &Weak<Process>| {
+            let current_process = process_ref.clone();
+            let sent_signal = move || {
+                let signal = KernelSignal::new(SIGALRM);
+                if let Some(process) = current_process.upgrade() {
+                    process.enqueue_signal(signal);
+                }
+            };
+
+            let work_sent_signal = {
+                let work_func = Box::new(sent_signal);
+                let work_item = Arc::new(WorkItem::new(work_func));
+                move || {
+                    submit_work_item(
+                        work_item.clone(),
+                        crate::thread::work_queue::WorkPriority::High,
+                    );
+                }
+            };
+
+            let realtime_timer_manager = id_to_global_manager(&ClockID::CLOCK_REALTIME).unwrap();
+            let real_timer = realtime_timer_manager.create_timer(work_sent_signal.clone());
+
+            let user_clock = Arc::new(CpuClock::new());
+            let kernel_clock = Arc::new(CpuClock::new());
+            let prof_clock = Arc::new(ProfClock::new(user_clock.clone(), kernel_clock));
+
+            let virtual_timer =
+                TimerManager::new(user_clock.clone()).create_timer(work_sent_signal.clone());
```

Comment:
> By this way, CpuClock will be totally internal details and does not need to be exposed

I move these details into `cpu_clock.rs` and make both `CpuClock` and `ProfClock` public (see the file modifications for more details). This is because the `CpuClock` is a more basic concept and `ProfClock` is actually a kind of `CpuClock`.

By this way, `ProfClock` can directly expose the user cpu clock and kernel cpu clock for usage, reducing the complexities of APIs.